es/index.js

```diff
@@ -174,6 +175,11 @@ export default function axiosRetry(axios, defaultOptions) {
   axios.interceptors.request.use(config => {
     const currentState = getCurrentState(config);
     currentState.lastRequestTime = Date.now();
+    const { adapter: origAdapter } = config;
+    config.adapter = function retryAdapter(config) {
```

Comment:
In particular this feels potentially problematic to me: if folks depend on config.adapter being the same instance of the adapter function that they pass in, this might break. 

Alternatively, mutating the adapter that consumers pass in also feels p gross, and actually causes errors since that means that different requests using the same axios instance will have share the same configuration (and hence retry counts). Happy to discuss this.

Reply:
Hi indeed we share the same point of view. We did a prototype that keeps the config on the instance itself but it would require to deprecate the possibility to pass `axios-retry` config as a second parameter.
i.e. we won't be able to support these kind of calls:
```javascript
client.get('http://example.com/test', {
  'axios-retry': {
    retries: 2
  }
})
```

Reply:
Actually, [this line](https://github.com/softonic/axios-retry/pull/76/files#diff-f92de460425d8a920c4aaed6bd0f6c44R91) prioritises instance-specific configs over the statically defined ones, so this continues to work (that specific test failed, which caused me to add the first part of the logical expression)